src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -68,7 +68,17 @@ public partial class Activity : IDisposable
         protected virtual void Dispose(bool disposing) { throw null; }
         public void SetCustomProperty(string propertyName, object? propertyValue) { throw null; }
         public object? GetCustomProperty(string propertyName) { throw null; }
-        public ActivityContext Context { get { throw null; } }
+        public System.Diagnostics.ActivityContext Context { get { throw null; } }
+        public System.Diagnostics.Activity.Enumerator<System.Collections.Generic.KeyValuePair<string, object?>> EnumerateTagObjects() { throw null; }
+        public System.Diagnostics.Activity.Enumerator<ActivityEvent> EnumerateEvents() { throw null; }
+        public System.Diagnostics.Activity.Enumerator<ActivityLink> EnumerateLinks() { throw null; }
+
+        public struct Enumerator<T>
+        {
+            public readonly System.Diagnostics.Activity.Enumerator<T> GetEnumerator() { throw null; }
```

Comment:
Is this just here to make the C# compiler happy when doing a foreach loop? It feels like we are abusing the Enumerable/Enumerator pattern a bit but as long as BCL review team was happy with it I am happy too : )

Reply:
> Is this just here to make the C# compiler happy when doing a foreach loop? 

Yes

> It feels like we are abusing the Enumerable/Enumerator pattern a bit but as long as BCL review team was happy with it I am happy too : )

Actually this was recommended by the BCL design reviewers :-)

Reply:
Just FYI here are a couple other examples of this being done that came up during the review:

https://github.com/dotnet/runtime/blob/1043f003c2b6e404014845e42d25513cebe2b9d9/src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs#L569

https://github.com/dotnet/runtime/blob/1043f003c2b6e404014845e42d25513cebe2b9d9/src/libraries/System.Private.CoreLib/src/System/Text/StringRuneEnumerator.cs#L25

Brings up a question, should we add...

```csharp
[ComponentModel.EditorBrowsable(ComponentModel.EditorBrowsableState.Never)]
```

...like `ChunkEnumerator` is doing?

Reply:
> Just FYI here are a couple other examples of this being done that came up during the review:

Actually, `StringRuneEnumerator` is implementing `IEnumerable<>`.

> Brings up a question, should we add...

Good point. Please go ahead and apply the attribute as we did in `StringBuilder`

Reply:
Updated

Reply:
Looks you need to update the ref file too.